index.bs

```diff
@@ -406,73 +413,154 @@ When this method is invoked, the user agent MUST execute the following algorithm
+        are thrown, set |rpId| to the value of `host` as computed by this procedure. Otherwise, return [=a promise rejected with=] a
         {{DOMException}} whose name is "{{SecurityError}}", and terminate this algorithm.
 
-4. Process each element of {{cryptoParameters}} using the following steps, to produce a new sequence |normalizedParameters|.
-    - Let |current| be the currently selected element of {{cryptoParameters}}.
-    - If `current.type` does not contain a {{ScopedCredentialType}} supported by this implementation, then stop processing
-        |current| and move on to the next element in {{cryptoParameters}}.
-    - Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a> [[!WebCryptoAPI]],
-        with |alg| set to `current.algorithm` and |op| set to 'generateKey'. If an error occurs during this
-        procedure, then stop processing |current| and move on to the next element in {{cryptoParameters}}.
-    - Add a new object of type {{ScopedCredentialParameters}} to |normalizedParameters|, with |type| set to `current.type` and
+    Issue(w3c/webauthn#259): The rest of this algorithm assumes |rpId| is an
+    [=origin=], but the above step sometimes produces a string.
+
+4. Let |normalizedParameters| be a new [=list=].
+5. [=list/For each=] |current| of {{cryptoParameters}}:
+    - If <code>|current|.{{ScopedCredentialParameters/type}}</code> does not
+        contain a {{ScopedCredentialType}} supported by this implementation,
+        then [=continue=].
+    - Let |normalizedAlgorithm| be the result of <a>normalizing an algorithm</a>
+        [[!WebCryptoAPI]], with |alg| set to
+        <code>|current|.{{algorithm}}</code> and |op| set to `"generateKey"`. If
+        an error occurs during this procedure, then [=continue=].
+    - [=list/Append=] a new object of type {{ScopedCredentialParameters}} to
+        |normalizedParameters|, with |type| set to
+        <code>|current|.{{ScopedCredentialParameters/type}}</code> and
         |algorithm| set to |normalizedAlgorithm|.
 
-5. If |normalizedAlgorithm| is empty and {{cryptoParameters}} was not empty, cancel the timer started in step 2, reject
-    |promise| with a DOMException whose name is "{{NotSupportedError}}", and terminate this algorithm.
+6. If |normalizedAlgorithm| is empty and {{cryptoParameters}} was not empty,
+    cancel the timer started in step 2, return [=a promise rejected with=] with
+    a {{DOMException}} whose name is "{{NotSupportedError}}", and terminate this
+    algorithm.
+
+7. Let |clientExtensions| be a new [=list=].
+8. If the {{ScopedCredentialOptions/extensions}} member of {{options}} is
+    <a>present</a>, then [=map/for each=] |extension| → |argument| of
+    <code>{{options}}.{{ScopedCredentialOptions/extensions}}</code>:
+    1. If |extension| is not supported by this client platform, [=continue=].
+    2. Let |result| be the result of running |extension|'s [=client processing=] algorithm on |argument|.
```

Comment:
see https://github.com/w3c/webauthn/pull/347/files#r103043381 for possible resolution...